metricflow/specs/query_parameter.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
```

Comment:
Will these (and the ones in query_interface.py) be accessed externally at all? If so, we might want to put these into the metricflow/protocols directory instead. That way we don't forget about them when we go through and establish API boundaries and clarify our contracts for what we expect and what might break at any time.